src/targets/cpu/lowering.cpp

```diff
@@ -421,25 +421,34 @@ MIGRAPHX_REGISTER_OP(cpu_im2col)
 struct max_pool
 {
     static std::string name() { return "max"; }
-    static double start() { return std::numeric_limits<double>::lowest(); }
+    template <class T>
+    static T start()
+    {
+        return std::numeric_limits<T>::lowest();
+    }
 
     static double apply(double x, double y)
     {
         double m = std::max(x, y);
         return (m);
     }
 
-    static double final(double x, double) { return (x); }
+    static double final(double x, std::size_t) { return (x); }
 };
 
 struct avg_pool
 {
     static std::string name() { return "average"; }
-    static double start() { return 0.0; }
+
+    template <class T>
+    static double start()
+    {
+        return 0.0;
+    }
 
     static double apply(double x, double y) { return x + y; }
 
-    static double final(double x, double y) { return x / y; }
+    static double final(double x, std::size_t y) { return (y == 0) ? x : (x / y); }
```

Comment:
Should this be `0` when `y == 0`? Or does this not matter?

Reply:
Yes, when y == 0, x is also 0.

Reply:
so, either way is fine.

Reply:
changed to return 0